### ISSUE TITLE
Import feature from mrpt-1.5 branch

### ISSUE THREAD

**Issue by jlblancoc** [MRPT/mrpt#894]
To do: 
import ("manual cherry-pick"): 
* [x]  cb4be3f95adf1443f6a2939ad7083b72e4dceb2e
* [x]  db01af47917a78802f2cedfb24e0fb02db548170
* [x]  4779d77627b2e908de303faf2cc1881af121028f  (added 2019-Jan-28)

**Comment by bear-my-wrath** [comment]
Hello, I'm an undergraduate and new to open source. I'd really like to be able to contribute to this organization, but I'm facing difficulty in finding inroads where I can begin. Could you guide me with solving this bug (or some easy bug)? I'd continue trying to solve other bugs once I understand the flow.

**Comment by jlblancoc** [comment]
Hi @bear-my-wrath !

Thanks for the offering :+1: 

Yes, this bug is an easy one, actually... You could start by checking out the [contributing guide](https://github.com/MRPT/mrpt/blob/master/.github/CONTRIBUTING.md). That list contains basically all you need to get started, hopefully. 

Then, there are many bugs labeled as "help wanted". They are all good starting points. This one in particular is easy: it's just a matter of getting the specific changes in those two commits (from the mrpt-1.5 branch), and merge them into one commit, on top of the master branch.

mrpt-1.5 is a branch where we keep an "old" / "stable" version of mprt, and where we only very ocasionally add new features (as it was this case). master is the "development" branch, towards the launch of mrpt 2.0.0 (the alpha is planned to happen within 1-2 months), so we need to "import" those features here too. 

Cheers

**Comment by bear-my-wrath** [comment]
@jlblancoc I wasn't successful :(
I made changes to my best intuition about the code, I learnt git commands to finally make a PR.
I'm trying to understand the CircleCI checks, "clang-format-lint - Failed"
Could you guide me with making a successful build?

**Comment by jlblancoc** [comment]
Just added a new commit (now they are three) to the list of commits to merge into master above...

**Comment by jlblancoc** [comment]
Ooops... forgot to close this one after #895.
